extensions/pkg/controller/bastion/reconciler.go

```diff
@@ -0,0 +1,195 @@
+	if err := r.updateStatusProcessing(ctx, bastion, operationType, "Deleting the bastion"); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	r.logger.Info("Starting the deletion of bastion", "bastion", bastion.Name)
+	r.recorder.Event(bastion, corev1.EventTypeNormal, EventBastionDeletion, "Deleting the bastion")
+
+	if err := r.actuator.Delete(ctx, bastion, cluster); err != nil {
+		msg := "Error deleting bastion"
+		r.recorder.Eventf(bastion, corev1.EventTypeWarning, EventBastionDeletion, "%s: %+v", msg, err)
+		_ = r.updateStatusError(ctx, extensionscontroller.ReconcileErrCauseOrErr(err), bastion, operationType, msg)
+		return extensionscontroller.ReconcileErr(err)
+	}
+
+	msg := "Successfully deleted bastion"
+	r.logger.Info(msg, "bastion", bastion.Name)
+	r.recorder.Event(bastion, corev1.EventTypeNormal, EventBastionDeletion, msg)
+	if err := r.updateStatusSuccess(ctx, bastion, operationType, msg); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	r.logger.Info("Removing finalizer.", "bastion", bastion.Name)
+	if err := controllerutils.RemoveFinalizer(ctx, r.reader, r.client, bastion, FinalizerName); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error removing finalizer from bastion: %+v", err)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) updateStatusProcessing(ctx context.Context, bastion *extensionsv1alpha1.Bastion, lastOperationType gardencorev1beta1.LastOperationType, description string) error {
+	return extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, r.client, bastion, func() error {
+		bastion.Status.LastOperation = extensionscontroller.LastOperation(lastOperationType, gardencorev1beta1.LastOperationStateProcessing, 1, description)
+		return nil
+	})
+}
+
+func (r *reconciler) updateStatusError(ctx context.Context, err error, bastion *extensionsv1alpha1.Bastion, lastOperationType gardencorev1beta1.LastOperationType, description string) error {
+	return extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, r.client, bastion, func() error {
+		bastion.Status.ObservedGeneration = bastion.Generation
+		bastion.Status.LastOperation, bastion.Status.LastError = extensionscontroller.ReconcileError(lastOperationType, gardencorev1beta1helper.FormatLastErrDescription(fmt.Errorf("%s: %v", description, err)), 50, gardencorev1beta1helper.ExtractErrorCodes(gardencorev1beta1helper.DetermineError(err, err.Error()))...)
+		return nil
+	})
+}
+
+func (r *reconciler) updateStatusSuccess(ctx context.Context, bastion *extensionsv1alpha1.Bastion, lastOperationType gardencorev1beta1.LastOperationType, description string) error {
+	return extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, r.client, bastion, func() error {
+		bastion.Status.ObservedGeneration = bastion.Generation
+		bastion.Status.LastOperation, bastion.Status.LastError = extensionscontroller.ReconcileSucceeded(lastOperationType, description)
+		return nil
+	})
+}
```

Comment:
See https://github.com/gardener/gardener/pull/3869 - maybe you can adapt this accordingly.